trieste/models/gpflux/models.py

```diff
@@ -111,7 +111,10 @@ def __init__(
                 f"received {type(self.optimizer.optimizer)} instead."
             )
 
-        self.original_lr = self.optimizer.optimizer.lr.numpy()
+        if not isinstance(
+            self.optimizer.optimizer.lr, tf.keras.optimizers.schedules.LearningRateSchedule
+        ):
+            self.original_lr = self.optimizer.optimizer.lr.numpy()
```

Comment:
Maybe add a one-line explanation of the potentially uninitialised instance variable? Also do we want a test with LearningRateSchedule and tf.function to make sure that compilation doesn't break with uninitialised instance variables (I don't think it does but it's probably good practice to check all our code paths under compilation).

Reply:
perhaps we can still initialise it to `None` in case of `LearningRateSchedule`?

Reply:
otherwise, how would I test that compilation doesn't break?

Reply:
You might be able to explicitly call tf.function on the optimize method in the test before executing it. But don't worry if you don't: I'm just starting to take a look at the way we handle compilation and as part of that I'll try to get a sense of what testing we're missing.